tests/test_multimodal_streaming.py

```
@@ -0,0 +1,732 @@
+    spec = build_image_token_spec(proc)
+    assert spec.image_token == "<|image|>"
+    assert spec.image_token_id == 258880
+
+
+def test_build_image_token_spec_gemma3_swaps_to_boi_token():
+    """Gemma-3: `image_token` is the post-expansion soft token; placeholder is `boi_token`."""
+    tok = _StubTokenizer(
+        {"<image_soft_token>": 262144, "<start_of_image>": 255999}
+    )
+    proc = _StubProcessor(
+        tok, image_token="<image_soft_token>", boi_token="<start_of_image>"
```

Ruff (0.15.11)

[error] Possible hardcoded password assigned to argument: "image_token"

(S106)